### DISCUSSION TITLE
Can't connect to database

### DISCUSSION THREAD

**Discussion by KBCACheia** [dani-garcia/vaultwarden#2134] [General]
After a restart the vaultwarden can't connect the database. And i get this message.

"[vaultwarden::util][WARN] Can't connect to database, retrying: Failed to create pool.
[CAUSE] Error (
None,
)"

Deployment environment
Debian 10
Docker
DB: PostgreSQL
Version: 1.23.0

![2021-12-04 (1)](https://user-images.githubusercontent.com/5051443/144723272-cc02e570-2917-460a-847a-166ca1aa0eb6.png)

**Comment by BlackDex** [comment]
Did you checked if the database is running and accessable?
You can also mark Vaultwarden to depend on the PostgreSQL container if you have that in the same compose file. That should at least ensure that PostgreSQL is started before Vaultwarden.

**Reply by KBCACheia** [reply]
> Did you checked if the database is running and accessable? You can also mark Vaultwarden to depend on the PostgreSQL container if you have that in the same compose file. That should at least ensure that PostgreSQL is started before Vaultwarden.

Hi, yes its running and others applications are accessing it.

It's a remote DB server, but not was changed on DB server since it's was working.

**Comment by BlackDex** [comment]
Are those other applications also run via docker? Is the ip changed our the hostname? Is there a firewall which could block the access?

**Reply by KBCACheia** [reply]
> Are those other applications also run via docker? Is the ip changed our the hostname? Is there a firewall which could block the access?

Yes, they are. No, everything is the same as before. There ins't firewall between this accesses (same network and both system are a clean debian).

**Reply by BlackDex** [reply]
And what if you restart the container later in the process?
Because if you haven't changed anything, that should indicate Vaultwarden also. Thus would make it strange.
Are you sure the config is correct? I see the config has beem saved via the admin interface, have you changed the compose file in between?

You could try to rename the `config.json` located in the data folder and see what happens.

**Reply by KBCACheia** [reply]
> And what if you restart the container later in the process? Because if you haven't changed anything, that should indicate Vaultwarden also. Thus would make it strange. Are you sure the config is correct? I see the config has beem saved via the admin interface, have you changed the compose file in between?
> 
> You could try to rename the `config.json` located in the data folder and see what happens.

I already stop, remove, pull a new image and nothing work.
The thing is the [CAUSE] Error (none,). I have no idea how troubleshooting with this information.

I was in vacation (hehe), there was a power failure and the server restart.  That's the only issue that happens between. I will try the config Json.

**Reply by BlackDex** [reply]
A power loss sounds like some config was either not saved correctly, or lost and now using the wrong one.

**Reply by KBCACheia** [reply]
> > And what if you restart the container later in the process? Because if you haven't changed anything, that should indicate Vaultwarden also. Thus would make it strange. Are you sure the config is correct? I see the config has beem saved via the admin interface, have you changed the compose file in between?
> > You could try to rename the `config.json` located in the data folder and see what happens.
> 
> I already stop, remove, pull a new image and nothing work. The thing is the [CAUSE] Error (none,). I have no idea how troubleshooting with this information.
> 
> I was in vacation (hehe), there was a power failure and the server restart. That's the only issue that happens between. I will try the config Json.

Renaming the config.json didn't work. I also try change the database credencial too.

**Reply by KBCACheia** [reply]
I notice the timezone of the container is different. I don't know if it was like this before, but i will try change it.

**Reply by KBCACheia** [reply]
> I notice the timezone of the container is different. I don't know if it was like this before, but i will try change it.

It didn't work.

**Comment by KBCACheia** [comment]
I figure it out. Even it was known by the host, the container was not recognizing the hostname (just the name) of DB server. Than i put the FQDN (with the domain [Its a internal network]) and the container could connect.

So:
hostname (didn't work)
hostname.domain.local (work).

But, before was working... 
Thanks a lot for your help @BlackDex
